command/troubleshoot/upstreams/troubleshoot_upstreams.go

```diff
@@ -0,0 +1,102 @@
+package upstreams
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/hashicorp/consul/command/flags"
+	troubleshoot "github.com/hashicorp/consul/troubleshoot/connect"
+	"github.com/mitchellh/cli"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	// flags
+	adminBind string
+}
+
+func (c *cmd) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	defaultAdminBind := "localhost:19000"
+	if adminBind := os.Getenv("ADMIN_BIND"); adminBind != "" {
+		defaultAdminBind = adminBind
+	}
+	c.flags.StringVar(&c.adminBind, "admin-bind", defaultAdminBind, "The address:port that envoy's admin endpoint is on.")
+
+	c.http = &flags.HTTPFlags{}
+	flags.Merge(c.flags, c.http.ClientFlags())
+	flags.Merge(c.flags, c.http.ServerFlags())
```

Comment:
Will this need to support Enterprise flags, e.g. Partitions? Sometimes these still go into OSS, e.g. [peering list](https://github.com/hashicorp/consul-enterprise/blob/eaca8f33f71da474d45dccddcbf1ce3f027b1ee4/command/peering/list/list.go#L46-L47).

Reply:
We don't have any enterprise-specific flags atm.

Reply:
OK. I imagine at some point you'll need them to query for services/catalog information in the correct partition/namespace. It might work on agent-ful consul because the client agent belongs to a partition.